rachelnavarro/step/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private ArrayList<TimeRange> possibleTimes = new ArrayList<TimeRange>();
```

Comment:
These two fields feel a bit off. The main method in this class is ```query``` which, to the user, looks like a stateless method. IE if you were to call it twice with the same exact arguments you should get the same results. However, due to these two fields, that's not necessarily the case. 

To fix this you might do one of the following:
1) every call to query zeroes out ```possibleTimes``` and ```currentTime```. This might work but it feels a bit hacky to me
2) you could instantiate possibleTimes and currentTime in query and pass it around. 
3) Something else perhaps iono im doing this review on a saturday morning